tests/Core/Generators/GeneratorTest.php

```diff
@@ -0,0 +1,240 @@
+            $exception = 'PHPUnit_Framework_Error';
+            $message   = 'processSniff() must be an instance of DOMNode, null given';
+        }
+
+        if (method_exists($this, 'expectExceptionMessage') === true) {
+            // PHPUnit 5.2.0+.
+            $this->expectException($exception);
+            $this->expectExceptionMessage($message);
+        } else {
+            // Ancient PHPUnit.
+            $this->setExpectedException($exception, $message);
+        }
+
+        $generator = new MockGenerator($ruleset);
+        $generator->generate();
+
+    }//end testGeneratingInvalidDocsResultsInException()
+
+
+    /**
+     * Verify the wiring for the generate() function.
+     *
+     * @param string $standard The standard to use for the test.
+     * @param string $expected The expected function output.
+     *
+     * @dataProvider dataGeneratingDocs
+     *
+     * @return void
+     */
+    public function testGeneratingDocs($standard, $expected)
+    {
+        // Set up the ruleset.
+        $config  = new ConfigDouble(["--standard=$standard"]);
+        $ruleset = new Ruleset($config);
+
+        // Make the test OS independent.
+        $expected = str_replace("\n", PHP_EOL, $expected);
+        $this->expectOutputString($expected);
+
+        $generator = new MockGenerator($ruleset);
+        $generator->generate();
+
+    }//end testGeneratingDocs()
+
+
+    /**
+     * Data provider.
+     *
+     * @return array<string, array<string, string>>
+     */
+    public static function dataGeneratingDocs()
+    {
+        $multidocExpected  = "No Content\n";
+        $multidocExpected .= "Code Comparison Only, Missing Standard Block\n";
+        $multidocExpected .= "One Standard Block, Code Comparison\n";
+        $multidocExpected .= "One Standard Block, No Code\n";
+        $multidocExpected .= "One Standard Block, Two Code Comparisons\n";
+        $multidocExpected .= "Two Standard Blocks, No Code\n";
+        $multidocExpected .= "Two Standard Blocks, One Code Comparison\n";
+        $multidocExpected .= "Two Standard Blocks, Three Code Comparisons\n";
+
+        return [
+            'Standard without docs'            => [
+                'standard' => __DIR__.'/NoDocsTest.xml',
+                'expected' => '',
+            ],
+            'Standard with one doc file'       => [
+                'standard' => __DIR__.'/OneDocTest.xml',
+                'expected' => "One Standard Block, No Code\n",
+            ],
+            'Standard with multiple doc files' => [
+                'standard' => __DIR__.'/StructureDocsTest.xml',
+                'expected' => $multidocExpected,
+            ],
+        ];
+
+    }//end dataGeneratingDocs()
+
+
+    /**
+     * Test that the documentation for each standard passed on the command-line is shown separately.
+     *
+     * @covers \PHP_CodeSniffer\Runner::runPHPCS
+     *
+     * @return void
+     */
+    public function testGeneratorWillShowEachStandardSeparately()
```

Comment:
Should this be moved to Runner tests once such tests are created? If I'm reading this correctly, this test is testing mostly this part of the Runner class:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/53306fa3f6e85c0711a1506df6f09b580f4c40dc/src/Runner.php#L91-L103

Reply:
Correct, it should at some point be moved. Same goes for a similar test in the `Ruleset\ExplainTest` file.